util/metrics/log_test.go

```diff
@@ -17,131 +17,226 @@ limitations under the License.
 package metrics
 
 import (
-	"github.com/sirupsen/logrus"
-	"github.com/sirupsen/logrus/hooks/test"
 	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/sirupsen/logrus"
 )
 
 func TestInt64Set(t *testing.T) {
-	metricName := "testMetric"
 	cases := []struct {
-		name       string
-		n          int64
-		fieldNames []string
-		fields     []string
-		want       string
-		err        bool
+		name   string
+		fields []string
+		sets   []map[int64][]string
+		want   map[string]map[string]interface{}
 	}{
 		{
 			name: "zero",
-			want: "int64 \"testMetric\".Set(0) = 0",
+			want: map[string]map[string]interface{}{},
+		},
+		{
+			name:   "basic",
+			fields: []string{"component"},
+			sets: []map[int64][]string{
+				{64: {"updater"}},
+			},
+			want: map[string]map[string]interface{}{
+				"component": {
+					"updater": mean{[]int64{64}},
+				},
+			},
 		},
 		{
-			name: "basic",
-			n:    3,
-			want: "int64 \"testMetric\".Set(3) = 3",
+			name:   "fields",
+			fields: []string{"component", "source"},
+			sets: []map[int64][]string{
+				{64: {"updater", "prow"}},
+			},
+			want: map[string]map[string]interface{}{
+				"component": {
+					"updater": mean{[]int64{64}},
+				},
+				"source": {
+					"prow": mean{[]int64{64}},
+				},
+			},
 		},
 		{
-			name:       "fields",
-			n:          3,
-			fieldNames: []string{"user"},
-			fields:     []string{"someone"},
-			want:       "int64 \"testMetric\".Set(3) = 3",
+			name:   "values",
+			fields: []string{"component"},
+			sets: []map[int64][]string{
+				{64: {"updater"}},
+				{32: {"updater"}},
+			},
+			want: map[string]map[string]interface{}{
+				"component": {
+					"updater": mean{[]int64{64, 32}},
+				},
+			},
 		},
 		{
-			name:       "not enough fields",
-			n:          3,
-			fieldNames: []string{"user"},
-			fields:     []string{},
-			err:        true,
+			name:   "values",
```

Comment:
nit: This is named the same as the last test case

Reply:
Done